blog/bloom-and-cuckoo.md

```diff
@@ -0,0 +1,158 @@
+scales linearly with the capacity. Cuckoo filters are more space-efficient. This result is consistent with the experiments reported in [1].
+
+<p align="center">
+<img src="./assets/bloom-cuckoo/capacity-size.png">
+</p>
+<p align="center">
+<i>Figure 1: We fix the false positive rate and initialize the filters with capacities ranging from 100K to 1M. The y-axis shows the size of the filters in Megabytes.</i>
+</p>
+
+Next, we fix the capacity and initialize the filters with false positive rates ranging from 0.0001 to 0.5. Fig. 2 shows that Cuckoo filters are more space-efficient.
+The gap between Counting Bloom filters and Cuckoo filters grows as the false positive rate decreases. This is also consistent with experiments in [1].
+
+<p align="center">
+<img src="./assets/bloom-cuckoo/fp-rate-size.png">
+</p>
+<p align="center">
+<i>Figure 2: We fix the capacity and initialize the filters with false positive rates ranging from 0.0001 to 0.5. The y-axis shows the size of the filters in Megabytes.</i>
+</p>
+
+## Lookup Performance
+We first add elements to both filters until the capacity is reached. We then measure the lookup performance for different ratios of positive and negative lookups.
+A positive lookup is for an existing element and a negative lookup is for an element that is not contained in the filter. For each ratio, we perform 100K lookups and report
+the average lookup duration and the standard deviation. Fig. 3 shows the results. Bloom filters perform slightly better on average than Cuckoo filters. This result is not consistent with [1],
+where Cuckoo filters were reported to have a better lookup performance compared to Bloom filters. It should be noted here that the authors in [1] use the original C++ Cuckoo filter implementation and their
+own unreleased Bloom filter implementation, while we use a Rust Cuckoo filter implementation and our Bloom filter implementation. We cannot easily determine the reason for this discrepancy,
+however, the performance difference is negligible.
+
+<p align="center">
+<img src="./assets/bloom-cuckoo/lookup.png">
+</p>
+<p align="center">
+<i>Figure 3: Lookup performance for different ratios of positive and negative lookups. For example, ratio 0.25 indicates that 25% of lookups are positive and 75% are negative.
+The shaded region indicates the standard deviation.</i>
+</p>
+
+## Insertion Performance
+Less critical than lookup performance but still important for our purposes is insertion performance. We measure how the insertion performance varies for different occupancy levels.
+Fig. 4 shows the results. The insertion performance is constant across all levels of occupancy
+for Bloom filters. For Cuckoo filters, the performance decreases as the filter becomes fuller because more relocations are required. Note that in Fig. 4, the performance for Bloom
+filters is actually not constant at first. It quickly increases and then remains constant. This can be explained by CPU caching. 
```

Comment:
```suggestion
for Bloom filters. For Cuckoo filters, the performance decreases as the filter becomes fuller because more relocations are required. In Fig. 4, the performance for Bloom filters is not constant. It quickly increases and then remains constant. This can be explained by CPU caching. 
```